### PR TITLE
RUM-3068 Add support for global attributes on logs

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What does this PR do?

Add a `Logs.addAttribute` and `Logs.removeAttribute` function to add attributes to all logs generated by Loggers created from a specific Core.  Individual loggers and log events override global attributes.

These log attributes are also sent on unhandled exceptions and added to errors forwarded to RUM.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

**Comment by fuzzybinary** [comment]
Creating as Draft while I work on getting the NDK to report the global attributes.

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/Logs.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/Logs.kt:57]:
```suggestion
     * Add a custom attribute to all future logs sent by loggers created from the given SDK core.
```

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/Logs.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/Logs.kt:85]:
```suggestion
     * Remove a custom attribute from all future logs sent by loggers created from the given SDK core.
```

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/Logs.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/Logs.kt:91]:
```suggestion
     * @param sdkCore the [SdkCore] instance to remove the attribute from. If not provided, the default
```

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/LogsFeature.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/LogsFeature.kt:74]:
```suggestion
    }

    /**
```

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/LogsFeature.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/LogsFeature.kt:88]:
```suggestion
    }
    
    // endregion
```

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/LogsFeature.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/LogsFeature.kt:64]:
```suggestion
    internal fun addAttribute(key: String, value: Any?) {
```

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/LogsFeature.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/LogsFeature.kt:80]:
```suggestion
    internal fun removeAttribute(key: String) {
```

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandler.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandler.kt:48]: since we are going to mutate it, let's make a copy

```suggestion
        val combinedAttributes = attributes.toMutableMap()
```

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandler.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandler.kt:52]: actually here we can simplify a bit, and considering my comment above, do something like:

```
val combinedAttributes = mutableMapOf<String, Any?>
val logsFeature = sdkCore.getFeature(Feature.LOGS_FEATURE_NAME)
if (logsFeature != null) {
  combinedAttributes.putAll(logsFeature.unwrap<LogsFeature>().getAttributes())
}
combinedAttributes.putAll(attributes)
```

Looks a bit cleaner and doesn't use `var`.

File: features/dd-sdk-android-logs/src/test/kotlin/com/datadog/android/log/LogsTest.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/test/kotlin/com/datadog/android/log/LogsTest.kt:99]:
```suggestion
    fun `M log user error W addAttribute { logs not enabled }`(
```

File: features/dd-sdk-android-logs/src/test/kotlin/com/datadog/android/log/LogsTest.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/test/kotlin/com/datadog/android/log/LogsTest.kt:124]:
```suggestion
    fun `M log user error W removeAttribute { logs not enabled }`(
```

File: features/dd-sdk-android-logs/src/test/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandlerTest.kt
Comment (0xnm) [inline features/dd-sdk-android-logs/src/test/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandlerTest.kt:336]:
```suggestion
    // region Forwarding to RUM
```

**Review by 0xnm — COMMENTED** [review]
nice job! I added a few comments, but overall lgtm.

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/LogsFeature.kt
Comment (fuzzybinary) [reply features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/LogsFeature.kt:74]: !? Merge conflict maybe?

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandler.kt
Comment (fuzzybinary) [reply features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandler.kt:52]: Agreed. I'll make the change.

**Review by 0xnm — APPROVED** [review]
lgtm! I added one comment. Let's wait for the iOS counterpart review before the merge to be aligned.

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/Logs.kt
Comment (mariusc83) [inline features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/Logs.kt:69]: should we accept `Any?` or just `Any` ?

**Review by mariusc83 — APPROVED** [review]

File: features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/Logs.kt
Comment (fuzzybinary) [reply features/dd-sdk-android-logs/src/main/kotlin/com/datadog/android/log/Logs.kt:69]: I kept it in line with Logger's interface which accepts `Any?`

**Pull request merged**
